pandas/tests/io/parser/comment.py

```diff
@@ -116,3 +118,29 @@ def test_commment_first_line(self):
         expected = DataFrame({0: ['a', '1'], 1: ['b', '2'], 2: ['c', '3']})
         result = self.read_csv(StringIO(data), comment='#', header=None)
         tm.assert_frame_equal(result, expected)
+
+    def test_comment_whitespace_delimited(self):
```

Comment:
You can use the `tm.capture_stderr` decorator from `pandas/util/testing.py` so you don't have to do the patching yourself. See https://github.com/pandas-dev/pandas/blob/6614e266bca32771c761e904367eff10dd4c8979/pandas/tests/io/parser/test_textreader.py#L146 for an example.